[Focus Mode] New images and content for the snippets article (#2876)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/javascript/snippets.md

   limitations under the License.  -->
# Run snippets of JavaScript on any webpage

> If you're running the same code in the **Console** tool repeatedly, consider saving the code as a snippet instead.  Snippets are scripts that you author in the **Sources** tool.  Snippets have access to the JavaScript context of the webpage and you can run snippets on any webpage.  Snippets can be used to alter a webpage, for example to change its content or appearance, or to extract data.

Reviewer: ```suggestion
If you're entering the same code into the **Console** tool repeatedly, consider saving the code as a snippet instead, and then running the snippet.  Snippets are scripts that you author in the **Sources** tool.  Snippets have access to the JavaScript context of the webpage and you can run snippets on any webpage.  Snippets can be used to alter a webpage, for example to change its content or appearance, or to extract data.
```
* "entering" - this is a more active, direct verb describing the labor.  "running" sounds more automatic/passive/abstract.
* ", and then running the snippet" - a more explicit, complete story of what we're proposing.

> The following screenshot shows Microsoft Edge with a webpage on the left, and DevTools on the right. The **Sources** tool is open showing the source code of a snippet. The snippet code was run, making changes to the webpage.

Reviewer: ```suggestion
The following screenshot shows Microsoft Edge with a webpage on the left and DevTools on the right. The **Sources** tool is open, displaying the source code of a snippet. The snippet code was run, making changes to the webpage:
```
rendered section: 
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/snippets?branch=pr-en-us-2876

![The webpage, altered by the snippet](./snippets-images/snippets-overview.png)

The snippet source code is shown below:

```javascript
// Change the background color to "red".
document.body.style.backgroundColor = "red";

// Add a paragraph at the bottom of the document.
const p = document.createElement("p");
p.textContent = "Hello world";
p.style.color = "white";
p.style.fontSize = "2rem";
document.body.appendChild(p);

// Log a message to the console.
console.log("Hello world");
```

The code changes the background color of the webpage to red, adds a new line of text at the bottom of the webpage, and logs a message to the **Console** tool.

> When you run a snippet on a webpage, the snippet's source code is added to the current webpage. For more information about changing the existing code of a webpage, instead of adding new code, see [Override webpage resources with local copies (Overrides tab)](./overrides.md).

Reviewer: ```suggestion
When you run a snippet on a webpage, the snippet's source code is added to the current webpage. For more information about changing the existing code of a webpage instead of adding new code, see [Override webpage resources with local copies (Overrides tab)](./overrides.md).
```
i like commas, wherever you'd pause while speaking.  But here imo, parses better w/o comma.  orig draft mis-parses as:

For info about X, don't do Y, do Z.
ie, with the comma, it's subject to mis-parsing as:
For more information about changing the existing code of a webpage, don't add new code; instead, see [Override webpage resources with local copies (Overrides tab)](./overrides.md).

> **Note:** The security settings of most webpages block from loading other scripts in snippets. For this reason, you must include all your code in one file.

Reviewer: ```suggestion

#### Include all your code in one file

The security settings of most webpages block from loading other scripts in snippets. For this reason, you must include all your code in one file.
```

The unstructured "Note: " construction is usually better replaced by an h4 serving like a skimmable "label" on the paragraph.  Heavy use of "Note" is a symptom of needing structured headings.

<!-- ====================================================================== -->
## Open the Snippets tab

> The **Snippets** tab is located in the **Navigator** pane, on the left of the **Sources** tool.

Reviewer: ```suggestion
The **Snippets** tab is located in the **Navigator** pane (grouped with the **Page** tab), on the left of the **Sources** tool.
```
i sometimes mention the default tab within a tool/panel, as a familiar reference that the user can look for.

To open the **Snippets** tab:

1. To open DevTools, right-click the webpage, and then select **Inspect**. Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS). DevTools opens.

1. In DevTools, on the Activity Bar, select the **Sources** tab. If that tab isn't visible, click the **More tools** (![More tools icon](./snippets-images/more-tools-icon.png)) button.

   ![The Sources tool with the Page tab open on the left](./snippets-images/sources-tool-page-pane.png)

1. In the **Navigator** pane (on the left), select the **Snippets** tab.  To access the **Snippets** option, you might need to click the **More tabs** (![More tabs](./snippets-images/more-tabs-icon.png)) button.

> #### Open the Snippets tab with the Command Menu

Reviewer: ```suggestion
#### Open the Snippets tab from the Command Menu
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with

> You can also open the **Snippets** tab with the Command Menu:

Reviewer: ```suggestion
You can also open the **Snippets** tab by using the Command Menu:
```

1. Select anything in DevTools, so that DevTools has focus.

1. Press **Ctrl+Shift+P** (Windows, Linux) or **Command+Shift+P** (macOS) to open the Command Menu.

> 1. Start typing "Snippets", select **Show Snippets**, and then press **Enter** to run the command:

Reviewer: ```suggestion
1. Start typing "snippets", select **Show Snippets**, and then press **Enter** to run the command:
```
realistically, in such a UI, i'd type "snippets", not "Snippets"

   ![The Show Snippets command](./snippets-images/command-menu.png)

<!-- ====================================================================== -->

> ## Create new snippets

Reviewer: ```suggestion
## Create a new snippet
```
the concrete task is creating _a_ snippet.  User creates 1 at a time.  generally, singular is clearer than plural; less ambiguous.

To create a new snippet from the **Snippets** tab:

1. [Open the Snippets tab](#open-the-snippets-tab).

> 1. Click **New snippet**.

Reviewer: change from:
1. Click **New snippet**.
to:
1. Click **New snippet** (**+**).
can use icon for the +

1. Enter a name for your snippet, and then press **Enter**:

   ![A new, empty, snippet in the Sources tool](./snippets-images/new-snippet.png)

> #### Create a new snippet with the Command Menu

Reviewer: ```suggestion
#### Create a new snippet from the Command Menu
```
"with" and "Note" are overused, like wildcards.
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with

1. Focus your cursor somewhere in DevTools.

1. Press **Ctrl+Shift+P** (Windows, Linux) or **Command+Shift+P** (macOS) to open the Command Menu.

1. Start typing "snippet", select **Create new snippet**, and then press **Enter**:

   ![The command for creating a new snippet](./snippets-images/new-snippet-command-menu.png)

> To rename your new snippet, see [Rename snippets](#rename-snippets).

Reviewer: ```suggestion
To rename your new snippet, see [Rename snippets](#rename-snippets), below.
```
Present section rendered:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/javascript/snippets?branch=pr-en-us-2876#create-a-new-snippet-with-the-command-menu

<!-- ====================================================================== -->

> ## Edit snippets

Reviewer: ```suggestion
## Edit a snippet
```
prefer singular to plural. The task/activity/scenario/ use-case is editing 1 snippet, not editing multiple snippets

To edit the source code of a snippet:

1. [Open the Snippets tab](#open-the-snippets-tab).

1. In the **Snippets** tab, click the name of the snippet that you want to edit.  The snippet opens in the **Code Editor**:

   ![The Code Editor](./snippets-images/edit-snippet.png)

1. Use the **Code Editor** to add JavaScript to your snippet.

1. When an asterisk appears next to the name of your snippet, it means you have unsaved code.  Press **Ctrl+S** (Windows, Linux) or **Command+S** (macOS) to save:

   ![An asterisk next to the snippet name indicates unsaved code](./snippets-images/unsaved-changes.png)

<!-- ====================================================================== -->

> ## Run snippets

Reviewer: ```suggestion
## Run a snippet
```
Global: prefer singular to plural.  The task/activity/scenario/ use-case is running 1 snippet, not running multiple snippets.

#### Run a snippet from the Sources tool

1. [Open the Snippets tab](#open-the-snippets-tab).

1. Click the name of the snippet that you want to run.  The snippet opens in the **Code Editor**:

1. Click the **Run snippet** (![Run snippet](./snippets-images/run-snippet-icon.png)) button, or press **Ctrl+Enter** (Windows, Linux) or **Command+Enter** (macOS):

   ![The run snippet button at the bottom of the code editor](./snippets-images/run-snippet-from-sources-tool.png)

#### Run a snippet with the Command Menu

1. Focus your cursor somewhere in DevTools.

1. Press **Ctrl+Shift+P** (Windows, Linux) or **Command+Shift+P** (macOS) to open the Command Menu.

1. Delete the **>** character and type the **!** character followed by the name of the snippet that you want to run:

   ![Running a snippet from the Command Menu](./snippets-images/run-snippet-from-command-menu.png)

1. Press **Enter** to run the snippet.

<!-- ====================================================================== -->

> ## Rename snippets

Reviewer: ```suggestion
## Rename a snippet
```

1. [Open the Snippets tab](#open-the-snippets-tab).

1. Right-click the snippet name, and then select **Rename**.

<!-- ====================================================================== -->

> ## Delete snippets

Reviewer: ```suggestion
## Delete a snippet
```

1. [Open the Snippets tab](#open-the-snippets-tab).

1. Right-click the snippet name, and then select **Remove**.

<!-- ====================================================================== -->

> ## Save snippets

Reviewer: ```suggestion
## Save a snippet
```

By default, snippets are only available within DevTools, but you can also save them to disk.

1. [Open the Snippets tab](#open-the-snippets-tab).

1. Right-click the snippet name, and then select **Save as**.

> 1. Enter a file name and location when prompted.

Reviewer: ```suggestion
1. When prompted, enter a file name and location.
```
even better might be construction:

1. Do x.  A Foo dialog opens.
2. Do Y.
